### PR TITLE
feat(python-cdk): add `description` to declarative source schema

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

This PR adds a `description` field to the `ManifestDeclarativeSource` schema, and tests that a source manifest with description parses as a valid source.

## WHY!?

The plan (between @lmossman, @bnchrch, and I) is to use this field to generate a general high level documentation blob for new manifest-only sources that people will make in Connector Builder.

While the platform will have it's own code to generate the documentation page, you can check out #42133 as an example of where this will surface, and #42132 for it's implemented in the CLI tools.

This by itself is a small change, once this lands, we will follow up with the Connector Builder frontend PR to actually write into this field.

## How

1. Adds the field in the schema yaml file.
2. `poetry run poe build` to generate the models
3. Test case change to make sure it's still valid.

## Review guide

- `.gitignore` change and `pyproject.toml` change to add formatting options with Ruff (my editor is using it by default) that work even if I open just the `airbyte_cdk/python` without causing problems for the repo, and without conflicting with our global Black formatting.

## User Impact

None.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/python/.gitignore [hunk 1]
```diff
@@ -2,3 +2,10 @@
 
 # TODO: these are tmp files generated by unit tests. They should go to the /tmp directory.
 <MagicMock*
+
+dist
+.ruff_cache
+.mypy_cache
+.venv
```
Comment (bnchrch) [inline airbyte-cdk/python/.gitignore:9]: 💎

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py [hunk 2]
```diff
@@ -37,8 +37,6 @@
 class ManifestDeclarativeSource(DeclarativeSource):
     """Declarative source defined by a manifest of low-code components that define source connector behavior"""
 
-    VALID_TOP_LEVEL_FIELDS = {"check", "definitions", "schemas", "spec", "streams", "type", "version"}
```
Comment (bnchrch) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py:40]: ❓ Were we not using these?

File: airbyte-cdk/python/pyproject.toml [hunk 3]
```diff
@@ -80,6 +80,12 @@ file-based = ["avro", "fastavro", "pyarrow", "unstructured", "pdf2image", "pdfmi
 sphinx-docs = ["Sphinx", "sphinx-rtd-theme"]
 vector-db-based = ["langchain", "openai", "cohere", "tiktoken"]
 
+[tool.ruff]
```
Comment (bnchrch) [inline airbyte-cdk/python/pyproject.toml:83]: 💎

**Review by bnchrch — APPROVED** [review]
One question. Nothing blocking. TY!

**Comment by natikgadzhi** [comment]
@lmossman you okay with naming? If yes, I will merge this on green CI, and then see if I can add an input in Builder as an exercise.

**Comment by lmossman** [comment]
@natikgadzhi so am I correct in understanding that the plan is to copy this description into both the connector doc file (i.e. `docs/integrations/connector-name.md`) and in the connector's README.md file whenever a connector is contributed?

What is the reasoning for adding this to the manifest itself as opposed to just setting the description on those files in the contribution endpoint?

**Comment by natikgadzhi** [comment]
@lmossman, hmmm, we could put the description directly in the docs blob, but then it's an extra thing to carry around and pass from frontend to the backend alongside the manifest.

I was thinking whether we should write the description into `metadata` _field_ of the manifest, but then thought that doesn't make much sense, as the inputs for descriptions will be rather prominent, and user-driven, not internal.

**Comment by lmossman** [comment]
Got it, I think it makes sense to try to push more things into the manifest rather than pass extra fields from the frontend so I'm on board with this.

And definitely agree that we should _not_ use the `metadata` field in the manifest for this, as that is pretty much reserved for frontend metadata at the moment

**Comment by natikgadzhi** [comment]
@maxi297 okay to merge with the failing cdk compat checks? I don't think they're relevant.

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py (hunk 2 above)
Comment (natikgadzhi) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py:40]: Yep, not actually found anywhere in the codebase.

**Comment by natikgadzhi** [comment]
@lmossman does the description field in the contribution modal use this field? 👀

**Comment by lmossman** [comment]
Not currently but it should, I will update that!

**Pull request merged**